tests/parser_factory_test.py

```diff
@@ -37,13 +37,33 @@ def test_get_parser() -> None:
 
 def test_vendors_in_readme() -> None:
     readme_path = Path(__file__).parent.parent.joinpath("README.md")
+    parsers: dict[ReleaseState, set[str]] = {
+        ReleaseState.RECOMMENDED: set(),
+        ReleaseState.CANDIDATE_RELEASE: set(),
+        ReleaseState.WORKING_DRAFT: set(),
+    }
+    # We don't include example parser in README
+    parsers[ReleaseState.WORKING_DRAFT].add(Vendor.EXAMPLE_WEYLAND_YUTANI.display_name)
+    section = None
     with open(readme_path) as f:
-        readme_contents = f.read()
+        for line in f:
+            if line.startswith("### Recommended"):
+                section = ReleaseState.RECOMMENDED
+            elif line.startswith("### Candidate Release"):
+                section = ReleaseState.CANDIDATE_RELEASE
+            elif line.startswith("### Working Draft"):
+                section = ReleaseState.WORKING_DRAFT
+            elif section and line.strip().startswith("-"):
+                parsers[section].add(line.strip()[2:])
+            else:
+                section = None
 
+    # Assert all vendors are in README
     for vendor in Vendor:
-        if vendor.release_state == ReleaseState.RECOMMENDED:
-            assert f"- {vendor.display_name}" in readme_contents
-        elif vendor.release_state == ReleaseState.CANDIDATE_RELEASE:
-            assert f"- *{vendor.display_name}" in readme_contents
-        else:
-            assert vendor.display_name not in readme_contents
+        assert vendor.display_name in parsers[vendor.release_state]
+
+    # Assert not extra parsers in README
+    assert (
+        set.union(*parsers.values()) - {vendor.display_name for vendor in Vendor}
```

Comment:
tiny nit: would prefer to check that the length of the result == 0 instead of comparing to `set()`

Reply:
This way actually gives a much more helpful error when the test fails.

Current version:

```
        )
E       AssertionError: assert {'Fake'} == set()
E
E         Extra items in the left set:
E         'Fake'
E         Use -v to get more diff

tests/parser_factory_test.py:66: AssertionError
================================================================================================================================== short test summary info ===================================================================================================================================
FAILED tests/parser_factory_test.py::test_vendors_in_readme - AssertionError: assert {'Fake'} == set()
```

With length check instead:

```
E       AssertionError: assert 1 == 0
E        +  where 1 = len(({'Agilent Gen5', 'Agilent Gen5 Image', 'AppBio AbsoluteQ', 'AppBio QuantStudio Design & Analysis', 'AppBio QuantStudio RT-PCR', 'Beckman PharmSpec', ...} - {'Agilent Gen5', 'Agilent Gen5 Image', 'AppBio AbsoluteQ', 'AppBio QuantStudio Design & Analysis', 'AppBio QuantStudio RT-PCR', 'Beckman PharmSpec', ...}))
E        +    where {'Agilent Gen5', 'Agilent Gen5 Image', 'AppBio AbsoluteQ', 'AppBio QuantStudio Design & Analysis', 'AppBio QuantStudio RT-PCR', 'Beckman PharmSpec', ...} = <method 'union' of 'set' objects>(*dict_values([{'Beckman Vi-Cell BLU', 'Beckman PharmSpec', 'Fake', 'AppBio AbsoluteQ', 'ChemoMetec Nucleoview', 'Bio-Ra...x xPONENT', 'Revvity Kaleido', 'PerkinElmer Envision'}, {'Qiacuity dPCR', 'Mabtech Apex'}, {'Example Weyland Yutani'}]))
E        +      where <method 'union' of 'set' objects> = set.union
E        +      and   dict_values([{'Beckman Vi-Cell BLU', 'Beckman PharmSpec', 'Fake', 'AppBio AbsoluteQ', 'ChemoMetec Nucleoview', 'Bio-Ra...x xPONENT', 'Revvity Kaleido', 'PerkinElmer Envision'}, {'Qiacuity dPCR', 'Mabtech Apex'}, {'Example Weyland Yutani'}]) = <built-in method values of dict object at 0x12da041c0>()
E        +        where <built-in method values of dict object at 0x12da041c0> = {<ReleaseState.RECOMMENDED: 'RECOMMENDED'>: {'Agilent Gen5', 'Agilent Gen5 Image', 'AppBio AbsoluteQ', 'AppBio QuantSt...ELEASE'>: {'Mabtech Apex', 'Qiacuity dPCR'}, <ReleaseState.WORKING_DRAFT: 'WORKING_DRAFT'>: {'Example Weyland Yutani'}}.values

tests/parser_factory_test.py:66: AssertionError
================================================================================================================================== short test summary info ===================================================================================================================================
FAILED tests/parser_factory_test.py::test_vendors_in_readme - AssertionError: assert 1 == 0
```